### PR TITLE
#3129354 Add Drupal 9 support

### PR DESCRIPTION
**Pull request by MPParsley**
https://www.drupal.org/project/og/issues/3129354 patch by @heddn.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,10 +12,10 @@
     },
     "require": {
         "php": ">=7.1.0",
-        "drupal/core": "~8.7"
+        "drupal/core": "~8.7 || ^9.0"
     },
     "require-dev": {
-        "drupal/coder": "~8.2"
+        "drupal/coder": "~8.2 | ^8.3"
```
Comment (MPParsley) [inline composer.json]: @heddn, shouldn't we use this:

```suggestion
        "drupal/coder": "^8.2"
```

**Comment by MPParsley** [comment]
@pfrenssen, did you plan another alpha release for Drupal 9?

File: composer.json (hunk 1 above)
Comment (heddn) [reply composer.json]: Yeah, better.

**Comment by pfrenssen** [comment]
Thanks for working on this! I see that the Drupal 9 builds on Travis are failing due to a composer conflict with Drupal Console.

It would be great if we can get the Drupal 9 build green in this PR. I had a look but there is no Drupal 9 compatible branch of Drupal Console yet.

Still let's move forward with this. I would propose to skip the installation of Drupal Console for Drupal 9 builds and mark the `DrupalConsoleAddFieldTest` as skipped. When Drupal Console becomes compatible we can re-enable this test.

**Review by pfrenssen — CHANGES_REQUESTED** [review]
Let's get the Drupal 9 builds green and remove them from the `allowed_failures` section in `.travis.yml`.

**Comment by amitaibu** [comment]
> Travis are failing due to a composer conflict with Drupal Console.

IMO we can remove integration with Drupal Console, thus remove Drupal Console completely.

**Comment by MPParsley** [comment]
> Let's get the Drupal 9 builds green and remove them from the `allowed_failures` section in `.travis.yml`.

Removed Drupal 9 from allowed_failures

**Comment by MPParsley** [comment]
> > Travis are failing due to a composer conflict with Drupal Console.
> 
> IMO we can remove integration with Drupal Console, thus remove Drupal Console completely.

Removed Drupal console from travis

**Comment by MPParsley** [comment]
@amitaibu 
Seems like there was another reason to use Drupal console, or did you suggest to remove the support (OgAddFieldCommand) all together?:

```
1) Drupal\Tests\og\Kernel\Console\DrupalConsoleAddFieldTest::testNewGroup
Error: Class 'Drupal\Console\Core\Style\DrupalStyle' not found
/home/travis/build/Gizra/og/src/Command/OgAddFieldCommand.php:80
/home/travis/build/Gizra/drupal/vendor/symfony/console/Command/Command.php:240
/home/travis/build/Gizra/drupal/vendor/symfony/console/Tester/CommandTester.php:78
/home/travis/build/Gizra/og/tests/src/Kernel/Console/DrupalConsoleAddFieldTest.php:75
ERRORS!
Tests: 562, Assertions: 4971, Errors: 1.
 > Executing tests from /home/travis/build/Gizra/drupal/modules/og/og_ui/tests
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.
Testing /home/travis/build/Gizra/drupal/modules/og/og_ui/tests

```

**Comment by amitaibu** [comment]
>  did you suggest to remove the support (OgAddFieldCommand) all together?

Indeed - can be removed. It's a very old attempt in adding Drupal Console commands, and I hardly think it's needed.

**Comment by MPParsley** [comment]
> > did you suggest to remove the support (OgAddFieldCommand) all together?
> 
> Indeed - can be removed. It's a very old attempt in adding Drupal Console commands, and I hardly think it's needed.

Rmeoved OgAddFieldCommand & DrupalConsoleAddFieldTest. Let's see what Travis says.

**Comment by MPParsley** [comment]
`PHP Fatal error:  Declaration of Drupal\Tests\Listeners\DrupalListener::endTest(PHPUnit\Framework\Test $test, float $time): void must be compatible with PHPUnit\Framework\TestListener::endTest(PHPUnit\Framework\Test $test, $time) in /home/travis/build/Gizra/drupal/core/tests/Drupal/Tests/Listeners/DrupalListener.php on line 14
`

**Comment by heddn** [comment]
I think the failures are due to PHPUnit version miss-matching.

File: .travis.yml [hunk 2]
```diff
@@ -63,10 +62,6 @@ before_script:
   # Remember the Drupal installation path.
   - DRUPAL_DIR=$(pwd)/drupal
 
-  # Download Drupal console so we can run the test for it. Skip this for the
-  # coding standards test.
-  - test ${TEST_SUITE} == "PHP_CodeSniffer" || composer require --dev drupal/console:~1.0 --working-dir=$DRUPAL_DIR
-
```
Comment (heddn) [inline .travis.yml:69]: I think we need to do some fancy logic and if TEST_SUITE == "9.0.x" then use phpunit 7.

File: .travis.yml [hunk 3]
```diff
@@ -63,16 +62,15 @@ before_script:
...
   # Install Composer dependencies for core. Skip this for the coding standards test.
   - test ${TEST_SUITE} == "PHP_CodeSniffer" || composer install --working-dir=$DRUPAL_DIR
 
   # PHPUnit 6 is required when running tests on PHP 7.x.
   - test ${TEST_SUITE} == "PHP_CodeSniffer" || test ${TRAVIS_PHP_VERSION:0:3} == "5.6" || composer require --dev phpunit/phpunit:~6 --update-with-dependencies --working-dir=$DRUPAL_DIR
 
+  # PHPUnit 7 is required when running tests Drupal 9.
+  - test ${TEST_SUITE:0:1} == "9"  || composer require --dev phpunit/phpunit:~7 --update-with-dependencies --working-dir=$DRUPAL_DIR
```
Comment (heddn) [inline .travis.yml]: Change the `||` to a `&&` and it should work.

Comment (MPParsley) [reply .travis.yml]:
```suggestion
  - test ${TEST_SUITE:0:1} == "9"  && composer require --dev phpunit/phpunit:~7 --update-with-dependencies --working-dir=$DRUPAL_DIR
```

**Comment by pfrenssen** [comment]
The D9 branch is green! :+1:

**Comment by pfrenssen** [comment]
I see what is wrong, I will try to fix it. We should remove the old workaround for PHP 5.6,

**Comment by pfrenssen** [comment]
Starting to see green results. I have also enabled the Composer cache, this saves ~30 seconds on downloading dependencies. I included Drupal 9.1 in the build matrix because this is also already in development.

Out of scope for this PR but I am seeing that on Drupal 8.8+ the only test that is failing is `OgComplexWidgetTest::testFields()`. This is great news, once that widget is removed we can also guarantee that OG works correctly on PHP 7.4.

**Comment by MPParsley** [comment]
And it's in! Thanks all!

**Comment by MPParsley** [comment]
@pfrenssen can you push it to d.o so we can close https://www.drupal.org/node/3129354 (only mobile access atm)?

**Comment by pfrenssen** [comment]
OK!

**Pull request merged**